embrace-android-features/src/main/kotlin/io/embrace/android/embracesdk/internal/injection/NativeFeatureModuleImpl.kt

```diff
@@ -95,6 +103,52 @@ internal class NativeFeatureModuleImpl(
         }
     }
 
+    override val nativeCrashHandlerInstaller: NativeCrashHandlerInstaller? by singleton {
+        if (configModule.configService.autoDataCaptureBehavior.isNativeCrashCaptureEnabled()) {
+            NativeCrashHandlerInstallerImpl(
+                configService = configModule.configService,
+                sharedObjectLoader = nativeCoreModule.sharedObjectLoader,
+                logger = initModule.logger,
+                repository = embraceNdkServiceRepository,
+                delegate = JniDelegateImpl(),
+                backgroundWorker = workerThreadModule.backgroundWorker(Worker.Background.IoRegWorker),
+                nativeInstallMessageProvider = {
+                    createNativeInstallMessage(
+                        storageModule.storageService,
+                        essentialServiceModule.sessionIdTracker.getActiveSessionId(),
+                        essentialServiceModule.processStateService.getAppState(),
+                        payloadSourceModule.deviceArchitecture.is32BitDevice
+                    )
+                },
+                mainThreadHandler = AndroidMainThreadHandler()
+            )
+        } else {
+            null
+        }
+    }
+
+    private fun createNativeInstallMessage(
```

Comment:
I'd suggest converting this to a val & adding a test case that checks it is populated with the correct values